entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java

```diff
@@ -45,16 +47,37 @@ public FilterTypeExpression convert(
     switch (operator) {
       case EQ:
         // 'field' EQ 'null' -> 'field' NOT_EXISTS
+        // array 'field' EQ 'null' -> 'field.valueList.values' NOT_EXISTS
         return RelationalExpression.of(
             identifierExpression, convertOperator(Operator.NOT_EXISTS), constantExpression);
 
       case NEQ:
         // 'field' NEQ 'null' -> 'field' EXISTS
+        // array 'field' NEQ 'null' -> 'field.valueList.values' EXISTS
         return RelationalExpression.of(
             identifierExpression, convertOperator(Operator.EXISTS), constantExpression);
 
       default:
         return relationalExpression;
     }
   }
+
+  /**
+   * An array is stored as 'attributes.array_field.valueList.values.%d.value.&lt;type&gt;'.
+   *
+   * <p>Example: products = ['shoes', 'socks'] is stored as
+   * 'attributes.names.valueList.values.0.value.string: shoes', and
+   * 'attributes.names.valueList.values.1.value.string: socks'.
+   *
+   * <p>An empty array is stored as 'attributes.array_field.valueList: {}'
+   */
+  private String getSubDocPath(String columnId, RequestContext requestContext)
```

Comment:
Once the above comment is addressed, we could get rid of these 2 methods as the same business logic is being done in org.hypertrace.entity.query.service.converter.identifier.IdentifierExpressionConverter